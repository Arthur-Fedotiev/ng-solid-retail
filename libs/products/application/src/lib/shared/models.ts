import { Categories, Colors, Retailers, Tiers } from './constants';

export type CategoriesMap = typeof Categories;
export type Category = typeof Categories[keyof typeof Categories];

export type RetailersMap = typeof Retailers;
export type TiersMap = typeof Tiers;
export type ColorsMap = typeof Colors;
