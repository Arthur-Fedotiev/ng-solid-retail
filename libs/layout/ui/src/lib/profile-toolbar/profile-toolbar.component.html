<mat-toolbar class="toolbar">
  <div class="toolbar__profile">
    <img
      [src]="
        'https://i.pinimg.com/280x280_RS/a9/7d/6f/a97d6f224a5526c8a49eedc19f0a73c4.jpg'
      "
      class="toolbar__avatar"
    />
    <span>John Doe</span>
    <button
      mat-icon-button
      [matMenuTriggerFor]="menu"
      aria-label="Open profile menu"
    >
      <mat-icon>keyboard_arrow_down</mat-icon>
    </button>
    <mat-menu #menu="matMenu">
      <button mat-menu-item>
        <mat-icon>dialpad</mat-icon>
        <span>Redial</span>
      </button>
      <button mat-menu-item disabled>
        <mat-icon>voicemail</mat-icon>
        <span>Check voice mail</span>
      </button>
      <button mat-menu-item>
        <mat-icon>notifications_off</mat-icon>
        <span>Disable alerts</span>
      </button>
    </mat-menu>
  </div>
</mat-toolbar>
