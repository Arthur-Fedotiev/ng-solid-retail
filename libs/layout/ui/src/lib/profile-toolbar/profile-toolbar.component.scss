.toolbar {
  display: flex;
  justify-content: space-between;

  &__profile {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-left: auto;
  }

  &__avatar {
    max-height: 3rem;
    border-radius: 1.5rem;
    margin: 0.25rem 1rem;
  }
}
