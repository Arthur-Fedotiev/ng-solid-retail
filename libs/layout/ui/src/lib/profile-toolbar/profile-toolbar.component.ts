import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'omnia-profile-toolbar',
  templateUrl: './profile-toolbar.component.html',
  styleUrls: ['./profile-toolbar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProfileToolbarComponent {}
