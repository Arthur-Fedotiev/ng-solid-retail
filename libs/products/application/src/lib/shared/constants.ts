import {
  ProductCategory,
  ProductRetailer,
  ProductTier,
  SHOES_AND_CLOTHING_COLORS,
} from '@sr/products/entities';

export const Categories = ProductCategory;
export const Retailers = ProductRetailer;
export const Tiers = ProductTier;
export const Colors = {
  [Categories.Shoes]: SHOES_AND_CLOTHING_COLORS,
  [Categories.Clothing]: SHOES_AND_CLOTHING_COLORS,
} as const;
