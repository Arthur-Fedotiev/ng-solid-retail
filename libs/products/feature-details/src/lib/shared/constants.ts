import { Tiers } from '@sr/products/application';

export const PRODUCT_TIERS = Tiers;
