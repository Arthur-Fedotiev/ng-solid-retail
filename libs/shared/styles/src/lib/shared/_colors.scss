$colors: (
  'M900': #1b1c1d,
  'M800': #4e5052,
  'M700': #969798,
  'M600': #f1f3f4,
  'M500': #727779,
  'M300': #d9dadc,
  'M100': #f1f3f4,
  'M0': #ffffff,
  'S50': #fafdfd,
  'S100': #f0f8f8,
  'B900': #4b79f7,
  'R50': #ffebee,
  'R100': #ffcacf,
  'R500': #fa3126,
  'R600': #eb2327,
  'R700': #d91421,
);

@function app-color($color) {
  @return map-get($colors, $color);
}
