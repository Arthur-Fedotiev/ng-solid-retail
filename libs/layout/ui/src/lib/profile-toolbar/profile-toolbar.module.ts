import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProfileToolbarComponent } from './profile-toolbar.component';
import { SharedUiMaterialModule } from '@sr/shared/ui-material';

@NgModule({
  declarations: [ProfileToolbarComponent],
  imports: [CommonModule, SharedUiMaterialModule],
  exports: [ProfileToolbarComponent],
})
export class ProfileToolbarModule {}
