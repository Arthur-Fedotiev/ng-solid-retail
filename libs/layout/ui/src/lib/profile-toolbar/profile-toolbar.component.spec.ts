import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SharedUiMaterialModule } from '@sr/shared/ui-material';

import { ProfileToolbarComponent } from './profile-toolbar.component';

describe('ProfileToolbarComponent', () => {
  let component: ProfileToolbarComponent;
  let fixture: ComponentFixture<ProfileToolbarComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SharedUiMaterialModule],
      declarations: [ProfileToolbarComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(ProfileToolbarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
