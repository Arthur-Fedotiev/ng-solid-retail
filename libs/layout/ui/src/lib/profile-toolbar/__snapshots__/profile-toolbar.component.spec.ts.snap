// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileToolbarComponent should render 1`] = `
<omnia-profile-toolbar
  __ngContext__={[Function Number]}
>
  <mat-toolbar
    class="mat-toolbar toolbar mat-toolbar-single-row"
  >
    <div
      class="toolbar__profile"
    >
      <img
        class="toolbar__avatar"
        src="https://lh3.googleusercontent.com/a-/AOh14GhmDJm8U1PpNgy_p9CsOeXrQkT8LEv7Mjv8woNdv08=s96-c"
      />
      <span>
        John Doe
      </span>
      <button
        aria-haspopup="menu"
        aria-label="Open profile menu"
        class="mat-focus-indicator mat-menu-trigger mat-icon-button mat-button-base"
        mat-icon-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
            data-mat-icon-type="font"
            role="img"
          >
            keyboard_arrow_down
          </mat-icon>
        </span>
        <span
          class="mat-ripple mat-button-ripple mat-button-ripple-round"
          matripple=""
        />
        <span
          class="mat-button-focus-overlay"
        />
      </button>
      <mat-menu>
      </mat-menu>
    </div>
  </mat-toolbar>
</omnia-profile-toolbar>
`;
